### ISSUE TITLE
How to avoid W002 warning?

### ISSUE THREAD

**Issue by PeSme** [AngellusMortis/django_microsoft_auth#251]
Django==2.2.3
django-microsoft-auth==2.3.0
Python 3.6.5

I think I have followed the docs but I still see 

> ?: (microsoft_auth.W002) `example.com` is still a valid site, Microsoft auth might not work
>         HINT: Microsoft/Xbox auth uses OAuth, which requires a real redirect URI to come back to
 
on any manage.py operation. The authentication works. But there is still this warning. I have just one site configured:

```
oncall=# select * from django_site;
 id |     domain     |      name
----+----------------+----------------
  1 | localhost:8000 | mysite
```
and my settings.py contains SITE_ID = 1. Removing it or setting it to None makes things worse:

```
  File "C:\Python36\lib\site-packages\microsoft_auth\apps.py", line 39, in microsoft_auth_validator
    Site.objects.get_current(request)
  File "C:\Python36\lib\site-packages\django\contrib\sites\models.py", line 60, in get_current
    return self._get_site_by_request(request)
  File "C:\Python36\lib\site-packages\django\contrib\sites\models.py", line 35, in _get_site_by_request
    host = request.get_host()
  File "C:\Python36\lib\site-packages\django\http\request.py", line 111, in get_host
    raise DisallowedHost(msg)
django.core.exceptions.DisallowedHost: Invalid HTTP_HOST header: 'example.com'. You may need to add 'example.com' to ALLOWED_HOSTS.
```

And setting ALLOWED_HOSTS = ['*'] is probably not what I want to do because of security, right?

Site.objects.get_current(request) prefers SITE_ID over provided request object, so 1 is returned and Site is fetched successfully. It's not 'example.com' but the warning is still displayed. Or if I do not set SITE_ID request.get_host() fails to validate_host() and that exception is not caught in the try-catch block in microsoft_auth_validator() where only Site.DoesNotExist is expected.

Am I missing something or is there a bug?

**Comment by AngellusMortis** [comment]
I think it is a bug. I am pretty sure that started happening after I added support for multiple sites. The warnings are designed to help you set everything up, if it is working, you can safely ignore it. Once I get a chance to look at it, I will clean up the warning so it is not always emitting.

**Comment by AngellusMortis** [comment]
Fixed in 2.3.1.
